theatre/shared/src/propTypes/utils.ts

```diff
@@ -28,3 +36,23 @@ export function getPropConfigByPath(
 
   return getPropConfigByPath(sub, rest)
 }
+
+/**
+ * @param value - An arbitrary value. May be matching the prop's type or not
+ * @param propConfig - The configuration object for a prop
+ * @returns value if it matches the prop's type, otherwise returns the default value for the prop
+ */
+export function valueInProp<PropValueType>(
+  value: unknown,
+  propConfig: IBasePropType<PropValueType>,
+): PropValueType {
+  const sanitize = propConfig.sanitize
+  if (!sanitize) return propConfig.default
```

Comment:
Hmm, we probably need to make `sanitize` mandatory. Otherwise, the prop's value would always be its default value. But that's for another PR.

Reply:
So I'd say that, for now, let's return the given value rather than the default value (which does match the assumptions made in `core/src/propTypes/index.ts`.

In another PR (which I'm working on), `sanitize()` will be mandatory, so this would not be a problem anymore.

Reply:
Thanks, this confirms and fills in my understanding. I am glad `sanitize` will be mandatory. I have modified `valueInProp` with this info in mind. I have also reverted some of my other extra typing changes so as not to encode extra assumptions before your PR is merged.